app/models/consultation.rb

```diff
@@ -87,6 +91,38 @@ def add_neighbour_addresses!(addresses, geojson)
     raise AddNeighbourAddressesError, e.message
   end
 
+  def polygon_fill_colour
+    "#{polygon_colour}20"
+  end
+
+  def concatenate_geojsons(*geojsons)
+    features = []
+
+    geojsons.compact.each do |geojson|
+      parsed = JSON.parse(geojson)
+
+      # Extract features based on whether the parsed object is a feature or a feature collection
+      if parsed["type"] == "Feature"
+        features << parsed
+      elsif parsed["type"] == "FeatureCollection" && parsed["features"].is_a?(Array)
```

Comment:
I'm a little wary of using `is_a?` here, but I don't really have a better suggestion so I don't think it's a blocker.

Could `parsed["features"]` ever be present but not an array? Maybe `parsed["features"].present?` would be sufficient?

Reply:
Method has been updated to `polygon_search_and_boundary_geojson`